test/Npgsql.Tests/TransactionTests.cs

```diff
@@ -46,9 +43,6 @@ public async Task Rollback([Values(PrepareOrNot.NotPrepared, PrepareOrNot.Prepar
             tx.Rollback();
             Assert.That(tx.IsCompleted);
             Assert.That(await conn.ExecuteScalarAsync($"SELECT COUNT(*) FROM {table}"), Is.EqualTo(0));
-            Assert.That(() => tx.Connection, Throws.Exception.TypeOf<InvalidOperationException>());
```

Comment:
Can probably leave a check that this does work, no? And definitely that accessing after dispose does throw (as above)?

Reply:
But it does not work. The best we can do is check for disposed.

Reply:
How do you mean, it doesn't work? Isn't the whole point of this PR to make NpgsqlTransaction.Connection work after the transaction has completed (rollback/commit), but before it's disposed?

Reply:
Oh, you've meant a check that it doesn't throw, correct? 🍞